Sources/Apollo/GraphQLResult.swift

```diff
@@ -1,15 +1,24 @@
+/// Represents source of data in GraphQLResult
+public enum GraphQLDataSource {
+    case cache
+    case server
+}
+
 /// Represents the result of a GraphQL operation.
 public struct GraphQLResult<Data> {
   /// The typed result data, or `nil` if an error was encountered that prevented a valid response.
   public let data: Data?
   /// A list of errors, or `nil` if the operation completed without encountering any errors.
   public let errors: [GraphQLError]?
-  
+  /// Source of data, or `nil` if there are any errors
+  public let dataSource:GraphQLDataSource?
```

Comment:
I don't think this should be optional. Even if there are errors, these are still part of the result and come from either the network or the cache.